src/ff.rs

```diff
@@ -103,9 +104,23 @@ impl FireflyAlgorithm {
               );
 
             for dimension in 0_usize..self.config.dimensions as usize {
-              population[index][dimension] += const1
-                * (population[innerindex][dimension] - population[index][dimension])
-                + self.config.alfa0 * alfa * (rng.gen_range(0.01..0.99)/*TODO ADD SETTING*/ - 0.5) * scale;
+              let step = const1 * (population[innerindex][dimension] - population[index][dimension])
+                + self.config.alfa0 * alfa * (rng.gen_range(0.01..0.99) - 0.5) * scale;
+              let _not_less_or_equal = matches!(
+                (population[index][dimension] + step).partial_cmp(&self.config.lower_bound),
+                None | Some(Ordering::Greater)
+              );
+              let _not_more_or_equal = matches!(
+                (population[index][dimension] + step).partial_cmp(&self.config.upper_bound),
+                None | Some(Ordering::Less)
+              );
+              if _not_more_or_equal && _not_less_or_equal {
+                population[index][dimension] += step;
+              } else if population[index][dimension] + step > self.config.upper_bound {
+                population[index][dimension] = self.config.upper_bound;
+              } else {
+                population[index][dimension] = self.config.lower_bound;
+              }
```

Comment:
```suggestion
              population[index][dimension] += step;
              population[index][dimension] = population[index][dimension].clamp(self.config.lower_bound, self.config.upper_bound);
```
Or maybe there is a reason for using `partial_cmp` instead of "<" and ">"

Reply:
Does num::clamp take undefineds into account?